### DISCUSSION TITLE
How to comment out a line in pages.jsonl

### DISCUSSION THREAD

**Discussion by eboston** [HASwitchPlate/openHASP#232] [Q&A] [answered]
I am trying to create a new page and running into some problems.  I am trying to narrow the problem down to which line is causing the problems.  I want to comment out all the lines, except one so I can see its behaviour, but I don't want to delete the lines I have there.  I have tried using "comment":"", but since the rest of the line is valid, it is still put into the display.

Is there an easy way to comment out lines?

**Comment by fvanroie** [comment]
JSON and JSONL format doesn't allow for commenting out lines... so that is not possible I'm affraid.

You can check the log via telnet or serial, it should say `"JSONL parsing failed at line x : reason"` when parsing fails.
You can see the objects being parsed in the log file, so it should be easy to find which line has the problem.

**Reply by eboston** [reply]
Well that's not good.  Any possibility we could get an object type of "comment" made to allow this?

As for the problem, it is not a parsing error.  It is with how objects, colors are getting displayed.

**Reply by fvanroie** [reply]
I think I misunderstood your question. What I meant was that JSON is a standard and you cannot comment out lines like this:

```json
// { "test" : "This line is not valid JSON" }
{ "test" : "This line is not valid JSON" } // comments here are not allowed
```

The `"comment"` key is only for human readable comments about that line of code to make it easier to organize and read the jsonl files. The comment value is completely ignored by the parser.

If the line is valid, but you still want to skip the parsing of the object maybe we can add a `"skip":true` key. Would that solve your problem?

**Reply by eboston** [reply]
I understand what you meant.  What I was suggesting would be to make an object called 'comment' that work like other objects such as 'btn', 'label', etc.  This object wouldn't do anything.

So by changing a line like this...

`{"id":4,"obj":"label","x":0,"y":2,"w":239,"h":65,"text":"\uE150","text_font":32,"align":"left"}`

to

`{"obj":"comment","id":4,"obj":"label","x":0,"y":2,"w":239,"h":65,"text":"\uE150","text_font":32,"align":"left"}`

the line would be ignored.

Does that make sense?

**Reply by fvanroie** [reply]
> Does that make sense?

It does, I finally got it. The problem is that's not a valid JSON line:

```
Error: Duplicate key 'obj'
```

So I would write it like this:

```json
{"skip":1,"id":4,"obj":"label","x":0,"y":2,"w":239,"h":65,"text":"\uE150","text_font":32,"align":"left"}
```

It's valid JSON and easy to spot and understand. If the `skip` key is not there we assume the line needs parsing.

**Reply by eboston** [reply]
Instead of "skip", could "comment" work?  Doesn't matter which word is used as long as the line is ignored.

**Reply by fvanroie** [reply]
yes, but `comment` already exists and means somehting else...
It would break existing examples and jsonl files

- `comment` = ignore this key only
- `skip` = ignore the entire object

**Reply by eboston** [reply]
Okay.  Makes sense.  Thanks.

**Comment by fvanroie** [comment, marked as answer]
I have pushed the change to the master branch and the development build can be downloaded from the [Github Action tab](https://github.com/HASwitchPlate/openHASP/actions/runs/1344250440) for testing.

**Reply by nagyrobi** [reply]
Is this in 0.6.2?
Is only thr key name monitored? Value can be arbitrary?

**Reply by fvanroie** [reply]
It is included in 0.6.2 but still needs testing.
If the key name `skip` exists **and** evaluates to `true` the line is completely skipped.
The value should be `0/1` or `false/true`.

If you can please confirm it is working as intended, we will add this to the documentation.

**Reply by nagyrobi** [reply]
Tested by adding `"skip":0` and `"skip":1` to various lines, also to various positions in the line, works. When 0 the object is not created. When 1 it is.

**Reply by fvanroie** [reply]
Thanks!
